src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleRuleService.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.activities
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.BadDataException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseScheduleRule
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.SlotCategory
+import java.time.DayOfWeek
+import java.time.LocalDateTime
+import java.time.temporal.TemporalAdjusters
+
+@Service
+class ScheduleRuleService {
+
+  fun mapRules(dto: CreateActivityRequest, courseActivity: CourseActivity) =
+    dto.scheduleRules.map {
+      validateRequest(it)
+
+      // Prod data usually uses 1st day of the scheduleStartDate month for the 'date' part
+      val monthStart = courseActivity.scheduleStartDate!!.with(TemporalAdjusters.firstDayOfMonth())
```

Comment:
Won't this mean that the data in Activities and Nomis might not match? Have yo uchecked this with Activities.

Also feels uncomfortable using !! here, could we take the date from the request instead?